### PR TITLE
Update Zones and VolumesPerServer to be string type 

### PR DESCRIPTION
**Pull request by nitisht**
string type is easier to use in scripting tools like
kustomize

### CODE REVIEW

File: pkg/apis/operator.min.io/v1/helper.go [hunk 1]
```diff
@@ -100,17 +101,19 @@ func (mi *MinIOInstance) AutoCert() bool {
 // VolumePath returns the paths for MinIO mounts based on
 // total number of volumes per MinIO server
 func (mi *MinIOInstance) VolumePath() string {
-	if mi.Spec.VolumesPerServer == 1 {
+	v, _ := strconv.Atoi(mi.Spec.VolumesPerServer)
```
Comment (kerneltime) [inline pkg/apis/operator.min.io/v1/helper.go:104]: We should not ignore the error here. Customers could be hand crafting the CRD.

File: pkg/resources/statefulsets/minio-statefulset.go [hunk 2]
```diff
@@ -385,7 +387,8 @@ func NewForMinIO(mi *miniov1.MinIOInstance, serviceName string, hostsTemplate st
 	if mi.Spec.VolumeClaimTemplate != nil {
 		pvClaim := *mi.Spec.VolumeClaimTemplate
 		name := pvClaim.Name
-		for i := 0; i < mi.Spec.VolumesPerServer; i++ {
+		v, _ := strconv.Atoi(mi.Spec.VolumesPerServer)
```
Comment (kerneltime) [inline pkg/resources/statefulsets/minio-statefulset.go:390]: Same here, we should validate the string.

File: pkg/resources/statefulsets/minio-statefulset.go [hunk 3]
```diff
@@ -121,13 +121,14 @@ func volumeMounts(mi *miniov1.MinIOInstance) (mounts []corev1.VolumeMount) {
 		name = mi.Spec.VolumeClaimTemplate.Name
 	}
 
-	if mi.Spec.VolumesPerServer == 1 {
+	v, _ := strconv.Atoi(mi.Spec.VolumesPerServer)
```
Comment (kerneltime) [inline pkg/resources/statefulsets/minio-statefulset.go:124]: Same here, we should validate the string.

Comment (nitisht) [reply pkg/resources/statefulsets/minio-statefulset.go:124]: validation is done initially here, https://github.com/minio/minio-operator/pull/169/files#diff-5fba95502c597c71c1ec5de3290ef01bR453 while someone submits a MinIO Tenant Yaml.

File: pkg/resources/statefulsets/minio-statefulset.go (hunk 2 above)
Comment (nitisht) [reply pkg/resources/statefulsets/minio-statefulset.go:390]: validation is done initially here, https://github.com/minio/minio-operator/pull/169/files#diff-5fba95502c597c71c1ec5de3290ef01bR453 while someone submits a MinIO Tenant Yaml.

File: pkg/apis/operator.min.io/v1/helper.go (hunk 1 above)
Comment (nitisht) [reply pkg/apis/operator.min.io/v1/helper.go:104]: validation is done initially here, https://github.com/minio/minio-operator/pull/169/files#diff-5fba95502c597c71c1ec5de3290ef01bR453 while someone submits a MinIO Tenant Yaml.

**Comment by nitisht** [comment]
Checking one more time if it is possible to use `int` with a custom variable in kustomize, blocked till then.

**Comment by nitisht** [comment]
> Checking one more time if it is possible to use `int` with a custom variable in kustomize, blocked till then.

`kustomize` currently supports int vars only for certain well known k8s objects like Deployment etc. For our use case, we need to have int vars for MinIO Tenant, which is not currently supports AFAICS. So we'll have to convert these fields to strings for now. 

Ref: https://github.com/kubernetes-sigs/kustomize/issues/1721
https://github.com/kubernetes-sigs/kustomize/issues/2052#issue-547209691

**Comment by nitisht** [comment]
With zone spec changes in #183 it is no longer possible to have zones as a string. Since kustomize doesn't support ints, we can avoid zones input via kustomize and let users use yamls for zone input for now. Closing this PR

**Pull request closed without merging**